components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/rabbitmq/RabbitMQConnectionConsumer.java

```diff
@@ -0,0 +1,368 @@
+        int retryInterval = rabbitMQConnectionFactory.getRetryInterval();
+        int retryCountMax = rabbitMQConnectionFactory.getRetryCount();
+        int retryCount = 0;
+        while ((workerState == STATE_STARTED) && !connection.isOpen()
+                && ((retryCountMax == -1) || (retryCount < retryCountMax))) {
+            retryCount++;
+            log.info("Attempting to reconnect to RabbitMQ Broker for the inbound " + factoryName +
+                    " in " + retryInterval + " ms");
+            try {
+                Thread.sleep(retryInterval);
+            } catch (InterruptedException e) {
+                log.error("Error while trying to reconnect to RabbitMQ Broker for the inbound " +
+                        factoryName, e);
+            }
+        }
+        if (connection.isOpen()) {
+            log.info("Successfully reconnected to RabbitMQ Broker for the inbound " + factoryName);
+            initConsumer();
+        } else {
+            log.error("Could not reconnect to the RabbitMQ Broker for the inbound " + factoryName +
+                    ". Connection is closed.");
+            workerState = STATE_FAULTY;
+        }
+    }
+
+    /**
+     * Used to start message consuming messages. This method is called in startup and when
+     * connection is re-connected. This method will request for the connection and create
+     * channel, queues, exchanges and bind queues to exchanges before consuming messages
+     *
+     * @throws ShutdownSignalException
+     * @throws IOException
+     */
+    private void startConsumer() throws ShutdownSignalException, IOException {
+        connection = getConnection();
+        if (channel == null || !channel.isOpen()) {
+            channel = connection.createChannel();
+            log.debug("Channel is not open. Creating a new channel for inbound " + factoryName);
+        }
+        //set the qos value for the consumer//
+        String qos = rabbitMQProperties.getProperty(RabbitMQConstants.CONSUMER_QOS);
+        if (qos != null && !"".equals(qos)) {
+            channel.basicQos(Integer.parseInt(qos));
+        }
+
+        //unable to connect to the queue
+        if (queueingConsumer == null) {
+            workerState = STATE_STOPPED;
+            return;
+        }
+
+        while (isActive()) {
+            try {
+                if (!channel.isOpen()) {
+                    channel = queueingConsumer.getChannel();
+                }
+                channel.txSelect();
+            } catch (IOException e) {
+                log.error("Error while starting transaction", e);
+                continue;
+            }
+
+            boolean successful = false;
+
+            RabbitMQMessage message = null;
+            try {
+                message = getConsumerDelivery(queueingConsumer);
+            } catch (InterruptedException e) {
+                log.error("Error while consuming message", e);
+                continue;
```

Comment:
AFAIU from the continue statement we are trying to retry the consumption of the message, it might be better to mention it in the error log.